builders/txBuilder.go

```diff
@@ -80,6 +88,66 @@ func (builder *txBuilder) ApplySignatureAndGenerateTx(
 	return builder.createTransaction(arg), nil
 }
 
+// ApplySignatureAndGenerateTxHash will sign the transaction and return it's hash
+func (builder *txBuilder) ApplySignatureAndGenerateTxHash(
```

Comment:
for reusability purposes: why not make convert the code a little bit, something like this function receives a parameter of type *data.Transaction and outputs its hash? In this way the function can be used also to compute a hash of a previously assembled transaction object (maybe from storage, from network, and so on)

Reply:
refactored